### PR TITLE
feat: Gemini SFT using multiple images and text

### PR DESCRIPTION
**Pull request by inardini**
# Description

This notebook demonstrates how to fine-tune the Gemini 1.5 Pro model for change detection task (spot differences) with the Vertex AI Supervised Tuning feature using multiple images and text as inputs.

Thank you for opening a Pull Request!
Before submitting your PR, there are a few things you can do to make sure it goes smoothly:

- [x] Follow the [`CONTRIBUTING` Guide](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/CONTRIBUTING.md).
- [x] You are listed as the author in your notebook or README file.
  - [x] Your account is listed in [`CODEOWNERS`](https://github.com/GoogleCloudPlatform/generative-ai/blob/main/.github/CODEOWNERS) for the file(s).
- [x] Make your Pull Request title in the <https://www.conventionalcommits.org/> specification.
- [x] Ensure the tests and linter pass (Run `nox -s format` from the repository root to format).
- [x] Appropriate docs were updated (if necessary)

### CODE REVIEW

File: gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb [hunk 1]
```diff
@@ -0,0 +1,924 @@
...
+        "    \"\"\"\n",
+        "    Random sample one image_id and its data from a jsonlines file.\n",
+        "    \"\"\"\n",
+        "    bucket_path = Path(input_test_file_path)\n",
+        "\n",
+        "    with bucket_path.open() as f:\n",
+        "        content = f.read()\n",
+        "        data = [\n",
+        "            json.loads(line) for line in content.strip().split(\"\\n\") if line.strip()\n",
+        "        ]\n",
```
Comment (holtskinner) [inline gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb:448]: Same as before, I recommend pandas

**Review by holtskinner — CHANGES_REQUESTED** [review]
Just a few comments

File: gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb [hunk 2]
```diff
@@ -0,0 +1,924 @@
...
+    },
+    {
+      "cell_type": "code",
+      "execution_count": null,
+      "metadata": {
+        "id": "iMbvf4daTif-"
+      },
+      "outputs": [],
+      "source": [
+        "def save_jsonlines(instances: list[dict[str, Any]], file_path: str) -> None:\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb:370]: tried but it does not generate the expected format which requires square brackets as part of the request

File: gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb [hunk 3]
```diff
@@ -0,0 +1,924 @@
...
+        "                \"parts\": [{\"text\": SYSTEM_INSTRUCTION}],\n",
+        "            },\n",
+        "            \"contents\": [\n",
+        "                {\n",
+        "                    \"role\": \"user\",\n",
+        "                    \"parts\": [\n",
+        "                        {\n",
+        "                            \"fileData\": {\n",
+        "                                \"mimeType\": \"image/png\",\n",
+        "                                \"fileUri\": f\"{image_path}.png\",\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb:417]: it is how the documentation requires to pass the call

File: gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb (hunk 1 above)
Comment (inardini) [reply gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb:448]: same as before

File: gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb [hunk 4]
```diff
@@ -0,0 +1,924 @@
...
+        "\n",
+        "def create_tuning_samples(file_path: str, split: str = \"train\") -> list[dict[str, Any]]:\n",
+        "    \"\"\"\n",
+        "    Creates tuning samples from a jsonlines file for image comparison tasks.\n",
+        "    \"\"\"\n",
+        "\n",
+        "    bucket_path = Path(file_path)\n",
+        "    instances = []\n",
+        "\n",
+        "    # Read and parse the jsonlines file\n",
```
[the remark this answers is not in the document]
Comment (inardini) [reply gemini/tuning/supervised_finetuning_using_gemini_on_multiple_images.ipynb:395]: same as below

**Review by gericdong — APPROVED** [review]

**Review by holtskinner — APPROVED** [review]

**Pull request merged**